www/js/recommendations.js

```diff
@@ -12,15 +12,33 @@ angular.module('emission.main.recommendations',['emission.services', 'emission.p
 
     $scope.name = "Cannot Retrieve Suggestion";
     $scope.mode = "Cannot Retrieve Mode";
+    $scope.bid = "yalis-stanley-hall-cafe-berkeley";
+    $scope.stars = 4.5;
+    $http.get('json/yelpfusion.json').then(function(result) {
+          $scope.yelp = result.data;
+        }
+    )
     $scope.clickX = function() {
       $scope.currentDisplay.style.display = "none";
     };
     $scope.clickReview = function(id) {
       $scope.currentDisplay = document.getElementById("reviewModal" + id);
       $scope.currentDisplay.style.display = "block";
+      $ionicLoading.show({
+        template: 'Loading Reviews...'
+        });
+      $http({
+        "async": true,
+        "crossDomain": true,
+        "url": "https://api.yelp.com/v3/businesses/"+$scope.bid+"/reviews",
+        "method": "GET",
+        "headers": $scope.yelp.headers
```

Comment:
this can technically have a race condition in which you try to read the yelp headers before they are initialized as part of the `$http` request above. I suspect this is hard to hit in your case because the header reading is nearly instantaneous, clearly faster than any human can press the review button.

I am going to merge this for now, but for the record, you should really understand asychronous programming and promises better :)